go/spotty/spotty_nix.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+
+package spotty
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"syscall"
+)
+
+func sameFile(ss1 *syscall.Stat_t, fi os.FileInfo) bool {
+	if ss2, ok := fi.Sys().(*syscall.Stat_t); ok {
+		return ss2.Dev == ss1.Dev && ss2.Rdev == ss1.Rdev && ss1.Ino == ss2.Ino
+	}
+	return false
+}
+
+// Discover which named TTY we have open as FD=0. Will return empty string or an error
+// if nothing was found, or if there was a problem, respectively. Will not work on Windows.
+func Discover() (string, error) {
+	var sstat syscall.Stat_t
+	if err := syscall.Fstat(0, &sstat); err != nil {
+		return "", err
+	}
+	res, err := findFileIn(&sstat, "/dev", regexp.MustCompile(`^tty[A-Za-z0-9]+$`))
+	if err != nil {
+		return "", err
+	}
+	if len(res) > 0 {
+		return res, nil
+	}
+	res, err = findFileIn(&sstat, "/dev/pts", regexp.MustCompile(`^[0-9]+$`))
+	return res, err
+}
+
+func findFileIn(ss *syscall.Stat_t, dir string, re *regexp.Regexp) (string, error) {
```

Comment:
Have you seen this func?  https://golang.org/pkg/path/filepath/#Walk

might help, not sure...